tests/unit/app/endpoints/test_ols.py

```diff
@@ -51,3 +65,132 @@ def test_nothing_is_logged_when_provider_and_model_not_set(self, caplog):
         # check captured outputs
         captured_out = caplog.text
         assert captured_out == ""
+
+
+@patch("ols.src.query_helpers.question_validator.QuestionValidator.validate_question")
+@patch("ols.src.query_helpers.docs_summarizer.DocsSummarizer.summarize")
+@patch("ols.src.query_helpers.yaml_generator.YamlGenerator.generate_yaml")
+@patch("ols.utils.config.conversation_cache.get")
+def test_conversation_request(
+    mock_conversation_cache_get,
+    mock_generate_yaml,
+    mock_summarize,
+    mock_validate_question,
+    load_config,
+):
+    """Test conversation request API endpoint."""
+    # valid question, no yaml
+    mock_validate_question.return_value = [
+        constants.SUBJECT_VALID,
+        constants.CATEGORY_GENERIC,
+    ]
+    mock_summarize.return_value = (
+        "Kubernetes is an open-source container-orchestration system...",  # summary
+        "",  # referenced documents
+    )
+    llm_request = LLMRequest(query="Tell me about Kubernetes")
+    response = ols.conversation_request(llm_request)
+    assert (
+        response.response
+        == "Kubernetes is an open-source container-orchestration system..."
+    )
+    assert len(response.conversation_id) > 0
+
+    # valid question, yaml
+    mock_validate_question.return_value = [
```

Comment:
it will be removed soon, please delete too

Reply:
Should I remove it here now? Wouldn't it be better to remove tests in the same PR where relevant logic is removed?